[fix] use C10_MOBILE/ANDROID/IOS

jerryzh168 opened the pull request:
Summary:
Didn't define C10_MOBILE in the numa file move diff: D13380559
move CAFFE2_MOBILE/ANDROID/IOS to c10

```
codemod -m -d caffe2 --extensions h,hpp,cc,cpp,mm "CAFFE2_MOBILE" "C10_MOBILE"
codemod -m -d caffe2 --extensions h,hpp,cc,cpp,mm "CAFFE2_ANDROID" "C10_ANDROID"
codemod -m -d caffe2 --extensions h,hpp,cc,cpp,mm "CAFFE2_IOS" "C10_IOS"

```

Differential Revision: D13490020

ezyang commented on c10/util/numa.cpp:
```diff
@@ -7,6 +7,9 @@ C10_DEFINE_bool(caffe2_cpu_numa_enabled, false, "Use NUMA whenever possible.");
 #define C10_ENABLE_NUMA
+#error "NUMA Enabled"
+#else
+#error "NUMA not Enabled"
```
I don't think you meant to commit this

jerryzh168 committed:
mobile regression (#15363)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/15363

Didn't define C10_MOBILE in the numa file move diff: D13380559
move CAFFE2_MOBILE/ANDROID/IOS to c10

```
codemod -m -d caffe2 --extensions h,hpp,cc,cpp,mm "CAFFE2_MOBILE" "C10_MOBILE"
codemod -m -d caffe2 --extensions h,hpp,cc,cpp,mm "CAFFE2_ANDROID" "C10_ANDROID"
codemod -m -d caffe2 --extensions h,hpp,cc,cpp,mm "CAFFE2_IOS" "C10_IOS"

```

i-am-not-moving-c2-to-c10

Differential Revision: D13490020

fbshipit-source-id: 77333b189e59e65f217c78aba62fe5bfc4a80b4b

jerryzh168 commented:
The regression is fixed. But the change of CAFFE2_MOBILE -> C10_MOBILE is still needed.
